src/hooks/useFlag.js

```diff
@@ -1,8 +1,20 @@
-// TODO: Write useFlag hook
+import compareAnswers from "../utils/compareAnswers";
+import { setFlagTrue } from "../redux/missions";
+import { useSelector, useDispatch } from "react-redux";
+import { addRecord } from "./useTerminal";
 
-// checkFlag: (state, { value, missionId, flagId }) => {
-//   // 1. Get mission and flag
-//   // 2. Hash value with md5
-//   // 3. Check if value is correct with saved falg
-//   // 4. If correct, set "done" to true
-// },
+const useFlag = (missionIndex, flagIndex, userAnswer) => {
+  const dispatch = useDispatch();
+  const savedAnswer = useSelector(
+    (state) => state.missions.list[missionIndex].flags[flagIndex].answer
+  );
+
+  if (compareAnswers(userAnswer, savedAnswer)) {
+    dispatch(setFlagTrue(missionIndex, flagIndex, answer));
+    addRecord({ value: `Provided flag is <b>correct<b>`, type: "output" });
+  } else {
+    addRecord({ value: `Provided flag is <b>incorrect<b>`, type: "output" });
+  }
+};
```

Comment:
should return at least {checkFlag} 
```suggestion
  return { checkFlag }
};
```